R/jwt.R

```diff
@@ -34,18 +35,25 @@
 #' pubkey <- as.list(mykey)$pubkey
 #' sig <- jwt_encode_sig(token, mykey)
 #' jwt_decode_sig(sig, pubkey)
-jwt_encode_hmac <- function(claim = jwt_claim(), secret, size = 256) {
+jwt_encode_hmac <- function(claim = jwt_claim(), secret, size = 256, header = NULL) {
   stopifnot(inherits(claim, "jwt_claim"))
   if(is.character(secret))
     secret <- charToRaw(secret)
   if(!is.raw(secret))
     stop("Secret must be a string or raw vector")
   if(inherits(secret, "rsa") || inherits(secret, "dsa") || inherits(secret, "ecdsa"))
     stop("Secret must be raw bytes, not a: ", class(secret)[-1])
-  header <- to_json(list(
-    typ = "JWT",
-    alg = paste0("HS", size)
-  ))
+  header <- if (is.null(header)) {
```

Comment:
I think if/else is redundant? If `header == NULL` then `c(list(...), header)` below will naturally reduce to just `list(...)` ?

Reply:
You are right